homeassistant/components/google_cloud/tts.py

```diff
@@ -0,0 +1,236 @@
+
+    @property
+    def default_language(self):
+        """Return the default language."""
+        return self._language
+
+    @property
+    def supported_options(self):
+        """Return a list of supported options."""
+        return SUPPORTED_OPTIONS
+
+    @property
+    def default_options(self):
+        """Return a dict including default options."""
+        return {
+            CONF_GENDER: self._gender,
+            CONF_VOICE: self._voice,
+            CONF_ENCODING: self._encoding,
+            CONF_SPEED: self._speed,
+            CONF_PITCH: self._pitch,
+            CONF_GAIN: self._gain,
+            CONF_PROFILES: self._profiles
+        }
+
+    async def async_get_tts_audio(self, message, language, options=None):
+        """Load TTS from google."""
+        _gender = options.get(CONF_GENDER).upper()
+        if _gender not in texttospeech.enums.SsmlVoiceGender.__members__:
+            _gender = self._gender
+
+        _voice = options.get(CONF_VOICE) or self._voice
+        if not re.match(VOICE_REGEX, _voice):
+            _voice = self._voice
+        if _voice and not _voice.startswith(language):
+            language = _voice[:5]
+
+        _encoding = options.get(CONF_ENCODING).upper()
+        if _encoding not in texttospeech.enums.AudioEncoding.__members__:
+            _encoding = self._encoding
+
+        _speed = options.get(CONF_SPEED)
+        _pitch = options.get(CONF_PITCH)
+        _gain = options.get(CONF_GAIN)
+        _profiles = options.get(CONF_PROFILES)
+
+        try:
+            # pylint: disable=no-member
+            synthesis_input = texttospeech.types.SynthesisInput(
+                text=message
+            )
+
+            voice = texttospeech.types.VoiceSelectionParams(
+                language_code=language,
+                ssml_gender=texttospeech.enums.SsmlVoiceGender[_gender],
+                name=_voice
+            )
+
+            audio_config = texttospeech.types.AudioConfig(
+                audio_encoding=texttospeech.enums.AudioEncoding[_encoding],
+                speaking_rate=max(min(_speed, MAX_SPEED), MIN_SPEED),
+                pitch=max(min(_pitch, MAX_PITCH), MIN_PITCH),
+                volume_gain_db=max(min(_gain, MAX_GAIN), MIN_GAIN),
+                effects_profile_id=_profiles or [],
+            )
+            # pylint: enable=no-member
+
+            with async_timeout.timeout(10, loop=self.hass.loop):
+                response = await self.hass.async_add_executor_job(
+                    self._client.synthesize_speech,
+                    synthesis_input,
+                    voice,
+                    audio_config
+                )
+                return _encoding, response.audio_content
+
+        except asyncio.TimeoutError as ex:
+            _LOGGER.error("Timeout for Google Cloud TTS call: %s", ex)
+        except Exception as ex:  # pylint: disable=broad-except
+            _LOGGER.error("Error occured during Google Cloud TTS call: %s", ex)
```

Comment:
use `_LOGGER.exception` to log stack trace.

Reply:
Thanks